dk.sdu.se4.player/src/main/java/dk/sdu/se4/player/PlayerProcessor.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package dk.sdu.se4.player;
+
+import dk.sdu.se4.common.entity.Entity;
+import dk.sdu.se4.common.entity.part.MovingPart;
+import dk.sdu.se4.common.service.ProcessorService;
+import dk.sdu.se4.commongameinput.GameInput;
+import dk.sdu.se4.commongameinput.GameInputKeys;
+
+/**
+ *
+ * @author Kasper
+ * @author Lucas
+ */
+public class PlayerProcessor extends PlayerCore implements ProcessorService {
+    
+    private GameInput gameInput = null;
+    
+    public void addGameInput(GameInput gameInput) {
+        this.gameInput = gameInput;
+    }
+    
+    public void removeGameInput(GameInput gameInput) {
+        this.gameInput = null;
+    }
+    
+    @Override
+    public void process() {
+        if(this.mapService != null) {
+            for(Entity e : this.mapService.getEntities(Player.class)) {
+                MovingPart mp = e.getPart(MovingPart.class);
+                mp.setMovingUp(false);
+                mp.setMovingDown(false);
+                mp.setMovingLeft(false);
+                mp.setMovingRight(false);
+                
+                if(this.gameInput.isPressed(GameInputKeys.UP)) {
+                    mp.setMovingUp(true);
+                } else if(this.gameInput.isPressed(GameInputKeys.DOWN)) {
+                    mp.setMovingDown(true);
+                }
+                if(this.gameInput.isPressed(GameInputKeys.LEFT)) {
+                    mp.setMovingLeft(true);
+                } else if(this.gameInput.isPressed(GameInputKeys.RIGHT)) {
+                    mp.setMovingRight(true);
+                }

```

Comment:
You don't need to do these checks because the moving part already does it. You should just have 4 lines of code that sets up, down, left and right directly using the inputs.